pytroll_collectors/tests/test_harvest_schedules.py

```diff
@@ -0,0 +1,257 @@
+            min_time, max_time = harvest_schedules(params, save_basename=self.basedir)
+            logs = str([rec.message for rec in self._caplog.records])
+            self.assertIsNone(min_time)
+            self.assertIsNone(max_time)
+            self.assertIn("Failed to download file:", logs)
+
+    def test_parse_schedules(self):
+        granule_metadata = {
+            'origin': '157.249.16.188:9063', 'sensor': ['viirs'], 'end_decimal': 3, 'stream': 'eumetcast',
+            'format': 'SDR_compact', 'orig_platform_name': 'npp',
+            'start_time': datetime.datetime(2019, 12, 16, 13, 44, 9), 'variant': 'EARS', 'orbit_number': 42154,
+            'dataset':
+            [{'uri':
+              '/data/pytroll/VIIRS-EARS/SVMC_npp_d20191216_t1344091_e1345333_b42154_c20191216135112000242_eum_ops.h5',
+              'uid': 'SVMC_npp_d20191216_t1344091_e1345333_b42154_c20191216135112000242_eum_ops.h5'}],
+            'start_decimal': 1, 'proctime': '20191216135112000242', 'antenna': 'ears', 'data_processing_level': '1B',
+            'tle_platform_name': 'SUOMI NPP', 'platform_name': 'suomi npp',
+            'end_time': datetime.datetime(2019, 12, 16, 13, 45, 33), 'type': 'HDF5', 'collection_area_id': 'eurol'}
+
+        planned_granule_times = set([datetime.datetime(2019, 12, 16, 13, 41, 18, 200000),
+                                     datetime.datetime(2019, 12, 16, 13, 42, 43, 600000),
+                                     datetime.datetime(2019, 12, 16, 13, 44, 9),
+                                     datetime.datetime(2019, 12, 16, 13, 45, 34, 400000),
+                                     datetime.datetime(2019, 12, 16, 13, 46, 59, 800000),
+                                     datetime.datetime(2019, 12, 16, 13, 48, 25, 200000),
+                                     datetime.datetime(2019, 12, 16, 13, 49, 50, 600000),
+                                     datetime.datetime(2019, 12, 16, 13, 51, 16),
+                                     datetime.datetime(2019, 12, 16, 13, 52, 41, 400000),
+                                     datetime.datetime(2019, 12, 16, 13, 54, 6, 800000),
+                                     datetime.datetime(2019, 12, 16, 13, 55, 32, 200000),
+                                     datetime.datetime(2019, 12, 16, 13, 56, 57, 600000)])
+        params = {'granule_metadata': granule_metadata,
+                  'planned_granule_times': planned_granule_times}
+        min_times, max_times = _parse_schedules(params, fake_test_pass_file.split("\n"))
+        self.assertEqual(min_times, datetime.datetime(2019, 12, 16, 13, 37))
+        self.assertEqual(max_times, datetime.datetime(2019, 12, 16, 14, 5))
+
+
+def suite():
+    """Test suite."""
+    loader = unittest.TestLoader()
+    mysuite = unittest.TestSuite()
+    mysuite.addTest(loader.loadTestsFromTestCase(TestHarvestSchedules))
+
+    return mysuite
+
+
+if __name__ == "__main__":
+    unittest.TextTestRunner(verbosity=2).run(suite())
```

Comment:
Since we are using pytest, this is not needed for automated testing at least.

Reply:
OK, I think I have copy/pasted some unittest framwork and mixed up with pytest.

Will try to clean up